plugins/aep-edge-mobile-track-hits/index.ts

```diff
@@ -27,40 +27,24 @@ import { ValidationPluginResult } from '../../types/validationPlugin';
     events
   ) as EdgeHitReceived[];
   const errors: string[] = [];
-
-  for (let i = 0; i < trackEvents.length; i++) {
-    let linked = false;
-    const trackEvent = trackEvents[i];
-    const identifier: string = aepMobile.mobileEvent.getEventId(trackEvent);
-
-    for (let j = 0; j < edgeHits.length; j++) {
-      const edgeHit = edgeHits[j];
-      const messages = toolkit.edge.edgeEvent.getMessages(edgeHit);
-      let data;
-      try {
-        data = messages ? JSON.parse(messages[1]) : val.payload;
-      } catch {
-        data = val.payload;
+  
+    for (let i = 0; i < trackEvents.length; i++) {
+      const trackEvent = trackEvents[i];
+      const identifier: string = aepMobile.mobileEvent.getEventId(trackEvent);
+      const linked = edgeHits.some(edgeHit => JSON.stringify(edgeHit).includes(identifier))
```

Comment:
I think we'll want to create a map of edge hit events with the parsed xdm message id prior to looping through the track events. That way we can quickly check if a track event has an edge hit without searching large strings of events. Something like:

```
const edgeHitMap = new Map();

for (const edgeHit of edgeHits) {
  const messages = toolkit.edge.edgeEvent.getMessages(edgeHit);
  const data = JSON.parse(messages[1]);
  const identifier = data.event.xdm._id;
  edgeHitMap.set(identifier, true);
}

We'll want to monitor the performance of this validator closely since large edge sessions will run this validator frequently.